File: scripts/make_counts_matrix.py
import pandas as pd

meta = pd.read_csv("../ref/mirna_practice_metadata.csv")
path = "../counts/"
samples = meta["sampleID"]

# Read in first sample
star_output = pd.read_table(f"{path}{samples.iloc[0]}.tsv", sep="\t", header=None)
# remove STAR output header lines 1-4 and take first two columns: 
# column 1 = gene ID, column 2 = unstranded counts
counts = star_output.iloc[4:, [0, 1]]

# Add in the rest of the samples
for i in range(1, len(samples)):
    star_output = pd.read_table(f"{path}{samples.iloc[i]}.tsv", sep="\t", header=None)
    star_output = star_output.loc[star_output[0].isin(counts[0].dropna()), :]
    counts = pd.concat([counts, star_output[1]], axis=1)

# Rename columns
counts.columns = ["gene_id"] + samples.tolist()

# Write to file
counts.to_csv("../counts/counts_matrix.tsv", sep="\t", index=False)

File: scripts/make_counts_matrix.R
meta = read.csv("../ref/mirna_metadata.csv")
path = "../counts/"
samples = meta$sampleID

# Read in first sample
star_output = read.table(paste0(path,samples[1],".tsv"),sep="",header=FALSE)
# remove STAR output header lines 1-4 and take first two columns: 
# column 1 = gene ID, column 2 = unstranded counts
counts = star_output[-c(1:4),c(1,2)]

# Add in rest of the samples
for(i in 2:length(samples)){
    star_output = read.table(paste0(path,samples[i],".tsv"),sep="",header=FALSE)
    star_output = star_output[na.omit(match(counts$V1,star_output$V1)),]
    counts=cbind(counts,star_output[,"V2"])
}

# Rename columns
colnames(counts) = c("gene_id",samples)

# Write to file
write.table(counts, file="../counts/counts_matrix.tsv", sep="\t", quote=F, row.names = F)
